src/libraries/Common/src/System/Net/Security/SecurityBuffer.Windows.cs

```diff
@@ -28,6 +29,62 @@ namespace System.Net.Security
         private SecurityBuffer _item2;
     }
 
+    [StructLayout(LayoutKind.Sequential)]
+    internal ref struct InputSecurityBuffers
+    {
+        internal int Count;
+        private InputSecurityBuffer _item0;
+        private InputSecurityBuffer _item1;
+        private InputSecurityBuffer _item2;
+
+        internal InputSecurityBuffer GetBuffer(int index)
+        {
+            Debug.Assert(index < 3);
+            return index == 0 ? _item0 : index == 1 ? _item1 : _item2;
+        }
+
+        internal void SetNextBuffer(InputSecurityBuffer buffer)
+        {
+            Debug.Assert(Count < 3);
+            if (Count == 0)
+            {
+                _item0 = buffer;
+            }
+            else if (Count == 1)
+            {
+                _item1 = buffer;
+            }
+            else
+            {
+                _item2 = buffer;
+            }
+
+            Count++;
+        }
+    }
+
+    [StructLayout(LayoutKind.Auto)]
```

Comment:
Why Auto? (It's unusual to see this attribute, so I'm curious)

Reply:
I copy that from the original `SecurityBuffer` as the intention was the same.
```c#
[StructLayout(LayoutKind.Auto)]
internal struct SecurityBuffer		     internal struct SecurityBuffer
```

Reply:
We generally use it when we may not have chosen the optimal layout for fields and want to allow the runtime to do so for us.